### DISCUSSION TITLE
Problems with logging drivers and quadlet?

### DISCUSSION THREAD

**Discussion by poperigby** [containers/podman#18316] [Q&A] [answered]
I recently updated to Fedora 38, and it seems like that gave me a new version of Podman (4.5). In that new version it looks like the default for Quadlet `.container` and `.kube` files is to have `--logging-driver=passthrough`. This makes it so I can't view container logs:

```
Error: this container is using the 'passthrough' log driver, cannot read logs: this container is not logging output
```

I thought to fix that, I would add `LoggingDriver=journald` under in `[Kube]` section of my `.kube` file. I then ran `systemctl --user daemon-reload`, and when I run `systemctl --user start example`, it says:

```
Failed to start caddy.service: Unit caddy.service not found.
```

What should I do? It seems weird that logging would be broken by default. Did I do something wrong?

**Comment by vrothberg** [comment]
> I recently updated to Fedora 38, and it seems like that gave me a new version of Podman (4.5). In that new version it looks like the default for Quadlet `.container` and `.kube` files is to have `--logging-driver=passthrough`. 

Commit 5de2a13e0928 actually reverted this to not set `passthrough` as the default and only adhere to system config.  You can double check by running `/usr/libexec/podman/quadlet -user -dryrun` and look if the log driver is set on the CLI.

> This makes it so I can't view container logs:
> 
> ```
> Error: this container is using the 'passthrough' log driver, cannot read logs: this container is not logging output
> ```
> 
> I thought to fix that, I would add `LoggingDriver=journald` under in `[Kube]` section of my `.kube` file. I then ran `systemctl --user daemon-reload`, and when I run `systemctl --user start example`, it says:
> 
> ```
> Failed to start caddy.service: Unit caddy.service not found.
> ```
> 
> What should I do? It seems weird that logging would be broken by default. Did I do something wrong?

Run `/usr/libexec/podman/quadlet -user -dryrun` and see what's wrong.  Probably Quadlet couldn't parse the file.

Can you also run `podman info|grep log` and see which log driver is set on your system?

**Comment by Luap99** [comment, marked as answer]
Are you sure you are using 4.5?

> Error: this container is using the 'passthrough' log driver, cannot read logs: this container is not logging output

I fixed this in 4.5 (https://github.com/containers/podman/pull/17502), this should no longer be displayed when you run podman logs on a quadlet unit.

**Comment by poperigby** [comment]
Thanks to you both! The new update fixed the logging problem. I'll keep the `/usr/libexec/podman/quadlet -user -dryrun` command in mind for the next time I have an issue with a Quadlet unit file.

I'm also no longer having the `Failed to start caddy.service: Unit caddy.service not found.` error when setting `LoggingDriver=journald`.
